src/main/java/com/siliconmtn/data/util/EntityUtil.java

```diff
@@ -54,6 +58,7 @@ public EntityUtil(EntityManager entityManager) {
 	 * @return an entity that was mapped by a dto
 	 */
 	public <T extends Object> T dtoToEntity(Object dto, Class<T> entity) {
+		if (dto == null) return null;
```

Comment:
Does entity also need checked here?  It's used with no checks below.

Reply:
The 'class' entity/dto I didn't check because I figured it would be provided with a given dto or entity, but the entity or dto itself could be null as it comes in.

Ex:
entityUtil.dtoToEntity(optionDTO, Option.class);

The class here is given and would never be null since it is needed, but if the object itself is null i just exit early as if an exception was thrown.

Reply:
i might do something weird later so I added the null handle on the class<?> inputs